examples/self_sustaining_parallel.py

```diff
@@ -0,0 +1,205 @@
+"""
+===========================================================
+Running a self-sustained flow of parallel tasks with joblib
+===========================================================
+
+This example introduces a recipe for implementing a feedback loop where the
+output generator of joblib.Parallel is used to fuel the input generator with
+new tasks, based on the value that are returned by the tasks that have been
+already completed, until some stopping criterion is met.
+
+The toy example given here starts by scheduling a few tasks that consists in
+incrementing an integer and returning it, then the feedback loop creates new
+tasks that increments again the returned integer, and so on, for as long as the
+integers remain below some threshold.
+
+The toy example has little value in itself, but introduces the skeleton of the
+implementation of this particular computational model, and it could be extended
+ to fulfill real world needs.
+
+For instance one such application is bayesian hyper-parameter search where the
+grid of parameters is explored based on a bayesian strategy, in which the
+cross-validated performance of hyper-parameter combinations that have been
+previously evaluated are used to choose the hyper-parameters that will be
+explored in the next rounds.
+
+"""
+
+
+##############################################################################
+# The following class implements the computational model. It can be easily
+# adapted to various usecases, since the functions that define the initial
+# tasks and the feedback loop are abstracted
+
+import joblib
+from queue import SimpleQueue
+
+
+class SelfSustainingParallelWork:
```

Comment:
Should this be called 
```suggestion
class ParallelSelfLoop:
```
?

I'm trying to find a name that makes this functionality as explicit as possible (not sure that "SelfLoop" is best thought, I'm really in search of ideas)